pages/_app.js

```diff
@@ -5,7 +5,27 @@ import { medications } from "@/lib/db";
 
 export default function App({ Component, pageProps }) {
   const [medicationsList, setMedicationsList] = useState(medications);
+  const [medicationPlan, setMedicationPlan] = useState([]);
+  const [forSomeoneElse, setForSomeoneElse] = useState(false);
+  const handleForChange = (event) => {
+    setForSomeoneElse(event.target.value === "Someone else");
+  };
 
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const frequency = [];
+    for (let [key, value] of formData.entries()) {
+      if (key.startsWith("frequency")) {
+        frequency.push(value);
+      }
+    }
+    const plan = {
+      ...Object.fromEntries(formData.entries()),
+      frequency,
+    };
+    setMedicationPlan((prevPlan) => [...prevPlan, plan]);
+  };
```

Comment:
since you are in your app it's not clear what is submitted here. Please use a more clear name…

```suggestion
  const handleAddMedicationPlanSubmit = (event) => {
    event.preventDefault();
    const formData = new FormData(event.target);
    const frequency = [];
    for (let [key, value] of formData.entries()) {
      if (key.startsWith("frequency")) {
        frequency.push(value);
      }
    }
    const plan = {
      ...Object.fromEntries(formData.entries()),
      frequency,
    };
    setMedicationPlan((prevPlan) => [...prevPlan, plan]);
  };
```

Reply:
Thank you for your feedback , i have already changed it